### PR TITLE
Convert project metadata to declarative setup.cfg

### PR DESCRIPTION
**Pull request by sirosen**
setuptools supports declarative metadata in setup.cfg, replacing the imperative style in setup.py. This results in a net-reduction in complexity for us, as we no longer have to maintain code for loading the readme or version data into the setuptools context.

Some specialized setuptools directives support pulling the version number from a package attribute and reading the description from the readme.

The rest of the metadata is a port of our setup.py content.

For broader compatibility with installations which explicitly invoke `setup.py`, a stub setup.py is still included, as is typical for packages which want to minimize the impact of a move to declarative metadata.

Two semantic changes are made here, on the premise that neither will be impactful for SDK users:
1. The vestigial 'dev' extra is removed. It contained no dependencies and there is no longer cause for anyone to install 'globus-sdk[dev]', which was purely for dev/testing purposes
2. The MANIFEST.in has been removed, which results in our tox.ini not being included in the built package distributions.

----
:books: Documentation preview :books:: https://globus-sdk-python--758.org.readthedocs.build/en/758/

### CODE REVIEW

File: MANIFEST.in
Comment (kurtmckee) [inline MANIFEST.in:1]: Does this still need to exist if it's empty? (Been a looooong time since I've dealt with this so it's not in my mind anymore...)

Comment (sirosen) [reply MANIFEST.in:1]: It does not need to exist. I had meant to remove it. Good catch that it's still in here!

(Aside, as I fix: you can see evidence of my first plan in my not-quite-right branch name. :grimacing: )

Comment (sirosen) [reply MANIFEST.in:1]: Oh, I misread. I thought you were saying that I had left it in as an empty file.
I realize now you were asking if we needed to include an empty one. I'm going to mark this thread resolved.

Comment (kurtmckee) [reply MANIFEST.in:1]: No, you read correctly. I really thought that it was an empty file, but now I see that GitHub says the file was deleted. Good to go!

**Review by kurtmckee — APPROVED** [review]

**Pull request merged**